### PR TITLE
[CHK-2904] change fields sent to notifications service

### PR DESCRIPTION
**Pull request by a-mannia**
#### List of Changes
Update the logic for populating the fields in the email template. For all non-card payments, based on the type of paymentTypeCode, the field values in the email template are changed with specific values.

#### Motivation and Context
Some fields were incorrectly populated in the receipt email template for specific paymentTypeCodes..

#### How Has This Been Tested?
Unit Test.
Tested in dev by simulating payments with different payment methods and verified the receipts.

#### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt [hunk 1]
```diff
@@ -82,67 +83,108 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
...
-              ?: transactionAuthorizationRequestData.authorizationRequestId,
-            // TODO: auth code is mandatory for OK mail template but not received for PAYPAL
-            // TODO task CHK-1970 will take cares of handle mail template for APM methods
-            transactionAuthorizationCompletedData.authorizationCode ?: "-",
-            PaymentMethodTemplate(
-              transactionAuthorizationRequestData.paymentMethodDescription,
-              transactionAuthorizationRequestData.transactionGatewayAuthorizationRequestedData.logo
-                .toString(),
+        when (transactionAuthorizationRequestData.paymentTypeCode) {
+          "PPAL",
```
Comment (petretiandrea) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt:98]: What about define the PaymentCode as a type or a set of constants? Something like
```kotlin
object PaymentCode {
    const val PAYPAL = "PPAL",
    const val MY_BANK = "MyBank"
    ...
}
```
or an enum

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt [hunk 2]
```diff
@@ -82,67 +83,108 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
...
+          "MyBank" ->
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt:99]: The payment type code for MyBank is `MYBK` (see [here](https://docs.pagopa.it/dizionario-dei-metadata/dati-canale-pagamento/mybank) `tipoVersamento`)
```suggestion
          "MYBK" ->
```

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt [hunk 3]
```diff
@@ -82,67 +83,108 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
...
+              createSuccessTemplate(
+                baseTransactionWithRequestedUserReceipt = baseTransactionWithRequestedUserReceipt,
+                emailAddress = emailAddress,
+                authorizationCode = transactionGatewayAuthorizationData.operationId)
+            } else {
+              throw RuntimeException(
+                "Unexpected TransactionGatewayAuthorization for paymentTypeCode ${transactionAuthorizationRequestData.paymentTypeCode}. Expected NPG gateway.")
+            }
+          }
+          "Redirect" ->
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt:117]: Also here, this corresponds to multiple payment type codes

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt [hunk 4]
```diff
@@ -82,67 +83,108 @@ class UserReceiptMailBuilder(@Autowired private val confidentialDataUtils: Confi
...
+            createSuccessTemplate(
+              baseTransactionWithRequestedUserReceipt = baseTransactionWithRequestedUserReceipt,
+              emailAddress = emailAddress,
+              authorizationCode = transactionAuthorizationCompletedData.authorizationCode ?: "-")
+          else -> {
```
Comment (giovanniberti) [inline src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt:122]: What cases are covered by this `else`?

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt (hunk 1 above)
Comment (a-mannia) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt]: I've created the Enum and used it in all the points of the code where it was needed:
https://github.com/pagopa/pagopa-ecommerce-event-dispatcher-service/blob/1a850cb5488801e12bd037bf6c3b7e7a1f46752c/src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/PaymentCode.kt#L1-L13

File: src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt (hunk 4 above)
Comment (a-mannia) [reply src/main/kotlin/it/pagopa/ecommerce/eventdispatcher/utils/v2/UserReceiptMailBuilder.kt]: I had a mistaken understanding of the data that the paymentTypeCode could take :')

**Review by giovanniberti — APPROVED** [review]

**Review by petretiandrea — APPROVED** [review]

**Pull request merged**